### PR TITLE
feat(proxy-mirror): support prefix mode

### LINKED ISSUE #8260: feat: As a user, I want to enhance the custom path of Proxy-mirror plugin, so that it can be used in more scenarios
### Description

ref: https://github.com/apache/apisix/issues/5305 https://github.com/apache/apisix/pull/6506

Under the current implementation, if `path` is set, the original request url will be replaced. We hope to add a `prefix` mode to adapt to more business scenarios.

for example:
Traffic under different domain names enters the unified traffic mirroring service and is differentiated by path
```
hostA/* -> Proxy-mirror  ->  mirroringService:8080/A/*
hostB/foo/* -> Proxy-mirror  ->  mirroringService:8080/B/foo/*
```

### PR DESCRIPTION
**Pull request by Gallardot**
### Description

Fixes #8260 

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/proxy-mirror.lua [hunk 1]
```diff
@@ -33,6 +33,12 @@ local schema = {
             type = "string",
             pattern = [[^/[^?&]+$]],
         },
+        mode = {
```
Comment (spacewander) [inline apisix/plugins/proxy-mirror.lua:36]: As it isn't "mode" of the proxy-mirror, I think calling it "path_concat_mode" would be better.

Comment (Gallardot) [reply apisix/plugins/proxy-mirror.lua]: @spacewander Done. PTAL.

**Review by spacewander — APPROVED** [review]
Could you update the doc (https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/proxy-mirror.md and the corresponding Chinese version) before we can merge it?

**Comment by Gallardot** [comment]
> Could you update the doc (https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/proxy-mirror.md and the corresponding Chinese version) before we can merge it?

@spacewander I have updated the documents in English and Chinese. PTAL.

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**